### PR TITLE
PCHR-2473: Error loading tasks and documents linked to an assignment

### PR DESCRIPTION
**Pull request by bmnepali**
## Overview
As of CiviCRM 4.7.22, when getting the assignments using AssignmentService, contacts were not fetched resulting the error in the console. This PR adds extra fields to be returned so that the contacts are available when fetching the assignments.

## Before
- Error in console due to no contacts in when getting assignments.
screen shot 2017-08-04 at 10 40 46 pm

## After
- No console error after the fix.
screen shot 2017-08-04 at 10 39 52 pm

## Technical Details
1. Added the `return` array with . list of fields to be returned when fetching the assignments in file  " uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js" in `get` function.
```js
        Assignment.get({
             'json': {
               'options': {
                  'limit': 0
                },
                'id': id,
                'return': ['case_type_id', 'contacts', 'client_id', 'contact_id', 'id', 'is_deleted', 'start_date', 'status_id', 'subject'],
                'debug': config.DEBUG
              }
            }, function (data) {
               ...
            }
```

## Comments
- As part of this PR, a prefix to a function in "uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/DAO/Assignment.php" is fixed.
```php
public static function $fields() {
    ...
}
```

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
-/* globals angular, CRM */
-/* eslint-env amd */
+/* eslint-env amd, jasmine */
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js]: this is not a spec file, hence it's not a jasmine environment

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 
 define([
+  'common/angular',
+  'common/lodash',
+  'tasks-assignments/services/contact',
   'tasks-assignments/services/services',
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js]:
```js
define([
  'common/angular',
  'common/lodash',
  'tasks-assignments/services/services',
  'tasks-assignments/services/contact',
  'tasks-assignments/services/utils'
], function (angular, _, services) {
});
```
you need to require `services/contact` but you don't need it as the callback parameter

Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js]: also, why do we need `angular`?

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js [hunk 3]
```diff
@@ -177,7 +180,6 @@ define([
         },
         updateCache: function (data) {
           $log.debug('updateCache');
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js]: remove this one also

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js [hunk 4]
```diff
@@ -65,6 +67,7 @@ define([
                 'limit': 0
               },
               'id': id,
+              'return': ['case_type_id', 'contacts', 'client_id', 'contact_id', 'id', 'is_deleted', 'start_date', 'status_id', 'subject'],
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js]: list the fields vertically

File: uk.co.compucorp.civicrm.tasksassignments/js/test/services/assignment_test.js [hunk 5]
```diff
@@ -36,5 +39,29 @@ define([
...
+            return: [ 'case_type_id', 'contacts', 'client_id', 'contact_id', 'id', 'is_deleted', 'start_date', 'status_id', 'subject' ],
+            debug: true
+          }
+        };
+
+        spyOn(Assignment, 'get').and.callThrough();
+        AssignmentService.get({ 'IN': [1, 2] });
+      });
+
+      it('calls Assignment.get with correct list of fields to be returned', function () {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/services/assignment_test.js]: you are just testing if the `return` field has a certain value, why do you need the `param` at all? You are just adding a log of noise that takes focus on what you are actually testing

```js
describe('get()', function () {
  beforeEach(function () {
    spyOn(Assignment, 'get').and.callThrough();
    AssignmentService.get({ 'IN': [1, 2] });
  });

  it('<title>', function () {
    expect(Assignment.calls.mostRecent().args[0]).toEqual(<array of fields>);
  });
});
```

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js (hunk 2 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js]: `angular` is being used in `updateCache` function in the `AssignmentService`.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js (hunk 4 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js]: will do.

**Comment by bmnepali** [comment]
@AkA84 I have already started the migration to new angular style guide for T&A extension in PR #239. Please check and let me know of improvements. Thanks.

File: uk.co.compucorp.civicrm.tasksassignments/js/test/services/assignment_test.js (hunk 5 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/services/assignment_test.js]: Refactored and implemented your solution.

**Pull request merged**